kanalony-storage-access/src/main/scala/kanalony/storage/generator/TableAccessorGenerator.scala

```diff
@@ -0,0 +1,115 @@
+package kanalony.storage.generator
+
+import kanalony.storage.generator.GenerationTemplates.{partitionKeyColumnDefinitionTemplate, clusteringKeyColumnDefinitionTemplate}
+
+/**
+ * Created by elad.benedict on 2/8/2016.
+ */
+class TableAccessorGenerator(val tm : TableMetadata) {
+
+  def generateClassName(): String = {
+    return TableAccessorGenerator.generateClassName(tm)
+  }
+
+  def generateEntityClassName(): String = {
+    EntityClassGenerator.getEntityName(tm)
+  }
+
+  def generateValuePopulation(): String = {
+    var res = "";
+    val columns = tm.columns
+    for (c <- columns){
+      var valueAssignment = GenerationTemplates.valueDefinitionTemplate.content
+      valueAssignment = valueAssignment.replace(GenerationTemplates.valueDefinitionTemplate.propertyNamePlaceholder, c.name)
+      res = res + valueAssignment + "\n"
+    }
+    res
+  }
+
+  def generateTableName(): String = {
+    tm.tableName
+  }
+
+  def generateColumnDefs[T <: IColumnDefinition](colDefs : List[T], templateCreator : T => IColumnDefinitionTemplate) : String = {
+    var res = "";
+    for (colDef <- colDefs){
+      val colDefTemplate = templateCreator(colDef)
+      val colDefResult = colDefTemplate.content.replace(colDefTemplate.propNamePlaceholder, colDef.name)
+        .replace(colDefTemplate.typePlaceholder, colDef.typeName.toString)
+      res = res + colDefResult + "\n"
+    }
+    res
+  }
+
+  def generatePartitionKeyDefs(colDefs : List[IColumnDefinition]) : String = {
+    generateColumnDefs[IColumnDefinition](colDefs, c => new partitionKeyColumnDefinitionTemplate())
+  }
+
+  def generateAdditionalColumnsDefs(colDefs : List[IColumnDefinition]) : String = {
+    generateColumnDefs[IColumnDefinition](colDefs, c => new GenerationTemplates.columnDefinitionTemplate())
+  }
+
+  def generateClusteringKeyDefs(colDefs : List[IClusteringColumnDefinition]) : String = {
+    generateColumnDefs[IClusteringColumnDefinition](colDefs, c => new GenerationTemplates.clusteringKeyColumnDefinitionTemplate(c.orderBy))
+  }
+
+  def generateTableColDefs(): String = {
+    var generatedContent = ""
```

Comment:
You probably wrote this for setting breakpoints. this can be written with no intermediates: 
`generatePartitionKeyDefs(tm.primaryKey.pk.columns) +`
`generateClusteringKeyDefs(tm.primaryKey.ck.columns) +` 
`generateAdditionalColumnsDefs(tm.additionalColumns)`
...and no need for semicolons